api/client/commands.go

```diff
@@ -227,6 +227,13 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 			return err
 		}
 	}
+
+	// windows: show error message about modified file permissions
+	if runtime.GOOS == "windows" {
```

Comment:
yeah I was thinking of adding a comment like 

// TODO(windows) should not display the warning when the daemon is windows

Checking the daemon os would be an API call, I hesitated to add something like that to a critical path like this one, I still do.